HW04.Rmd

```diff
@@ -1,126 +1,273 @@
+    theme_bw()
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+##make age variable numeric
+gun_death$age<- as.numeric(gun_death$age)
+
+#mean of female victims
+female_age<- gun_death %>%
+    select(age, sex) %>%
+    filter(sex=="F") 
+
+avg<- mean(female_age$age, na.rm=TRUE)  
+avg<- round(avg, digits=2)
+
+#plot
+ggplot(gun_death, aes(x=sex, y=age, fill=sex))+
+    geom_boxplot()+
+    annotate("text", x=1, y=70, label=paste("The average age\n of female victim is", avg))+
+    labs(title="Ages of Gun Violence Victims in 2012, By Gender", x=element_blank(), y="Age", fill="Gender")+
+    theme_bw()
+```
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+-The answer is 46227 
+
+```{r education}
+answer<- gun_death %>%
+    select(education, race, sex) %>%
+    filter(education %in% c("HS/GED", "Some college", "BA+"), race== "White", sex=="M") %>%
+    count()
+answer
+```
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+-Summer has the most gun deaths.
+
+```{r season}
+#calculate death count in each season
+winter<- gun_death %>%
+    select(month) %>%
+    filter(month %in% c("Jan", "Feb", "Mar")) %>%
+    count() 
+
+spring<- gun_death %>%
+    select(month) %>%
+    filter(month %in% c("Apr", "May", "Jun")) %>%
+    count() 
+
+summer<- gun_death %>%
+    select(month) %>%
+    filter(month %in% c("Jul", "Aug", "Sep")) %>%
+    count() 
+
+fall<- gun_death %>%
+    select(month) %>%
+    filter(month %in% c("Oct", "Nov", "Dec")) %>%
+    count() 
+
+max(winter, spring, summer, fall)
+
+```
+# These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
+
+### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
+
+```{r race}
+#data frame of gun deaths intent 
+gun_death_race3<- gun_death %>%
+    filter(race %in% c("White", "Black", "Hispanic"), intent %in% c("Suicide", "Homicide")) 
+
+#order race
+gun_death_race3$race<- factor(gun_death_race3$race, 
+                                 levels=c("White", "Black", "Hispanic")) 
+
+#plot
+ggplot(gun_death_race3, aes(x=race, fill=intent))+
+    geom_bar(position="stack", alpha=0.9)+
+    labs(title="Gun Death Intentions Comparisons Among Three Populations", x="Race", y="Number of deaths", fill="Reported Intentions")+
+    theme_bw()
+```
+
+(hint maybe looking at percentages would be good)
+```{r race_2}
+
+#add a percent column summarizing the info
+gun_death_percent<- gun_death_race3 %>%
+    group_by(race, intent) %>%
+    count() %>%
+    rename(death=n) %>%
+    group_by(race) %>%
+    mutate(total=sum(death), percent=death/total*100)
+
```

Comment:
The logic becomes very clear when using %>% to split each step! I haven't got used to this format